app/src/modules/extension/mv3RuleParser/parseQueryParamRule.ts

```diff
@@ -19,7 +19,10 @@ const parseQueryParams = (modifications: QueryParamRuleModification[]): QueryPar
     } else if (modification.type === QueryParamModificationType.REMOVE) {
       transform.queryTransform.removeParams.push(modification.param);
     } else {
+      // case remove all
       transform.query = "";
+      // queryTransform is not needed if query is present
+      delete transform.queryTransform;
```

Comment:
This won't work if a pair contains removeAll as well as other header modifications. This will only work if removeAll is the last modification